app/controllers/file.js

```diff
@@ -57,8 +58,20 @@ FileController.prototype.downloadFile = function(opts) {
       .on("response", (res) => {
         this.hasDownloadError = false;
         if (res.statusCode == 200) {
-          this.writeFile(res);
-          this.emit("downloading");
+
+          let fileSize = res.headers["content-length"];
+          let self = this;
+          fileSystem.isThereAvailableSpace(this.logger, function(isThereAvailableSpace) {
+            if(isThereAvailableSpace) {
+
+              fileSystem.addToDownloadTotalSize(fileSize);
+              self.writeFile(res);
+              self.emit("downloading");
+
+            } else {
+              self.emit("insufficient-disk-space", fileSize);
+            }
+          }, fileSize);
```

Comment:
@stulees it is passing in here. Last parameter of the function after a callback. cheers